projects/packages/woocommerce-analytics/src/class-wc-analytics-tracking.php

```diff
@@ -297,8 +297,21 @@ private static function get_visitor_id() {
 			return self::$cached_visitor_id;
 		}
 
-		self::$cached_visitor_id = null;
-		return null;
+		// Generate anonymous ID
+		self::$cached_visitor_id = wp_generate_password( 24 );
+		setcookie(
+			'tk_ai',
+			self::$cached_visitor_id,
+			array(
+				'expires'  => time() + ( 365 * 24 * 60 * 60 ), // 1 year
+				'path'     => '/',
+				'domain'   => COOKIE_DOMAIN,
+				'secure'   => is_ssl(),
+				'httponly' => true,
+				'samesite' => 'Strict',
+			)
+		);
```

Comment:
Should we gate the `setcookie` here? I know in Woo we specifically do a check if it is NOT a REST request: https://github.com/woocommerce/woocommerce/blob/bd757a86aa0df97a709ee9a50bbb6fc5372a0681/plugins/woocommerce/includes/tracks/class-wc-tracks-client.php#L74
```
if ( ! Constants::is_true( 'REST_REQUEST' ) && ! Constants::is_true( 'XMLRPC_REQUEST' ) ) {
			WC_Site_Tracking::set_tracking_cookie( 'tk_ai', $anon_id );
		}
```

Reply:
Good catch, updated the PR to check the REST_REQUEST and XMLRPC_REQUEST in the same way as the Jetpack code. I've also changed the wat to generate the anonymous ID to be more consistent with the Jetpack code